api/v1/nodescan/service.go

```diff
@@ -270,6 +272,15 @@ func (s *serviceImpl) GetNodeVulnerabilities(_ context.Context, req *v1.GetNodeV
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	// Handle the new format of the request and scan node inventory additionally
+	if features.RHCOSNodeScanning.Enabled() && req.GetNodeInventory() != nil {
+		// TODO(ROX-12968): resolve hardcoded value uncertifiedRHEL
+		layer, err := apiV1.GetVulnerabilitiesForComponents(s.db, req.GetNodeInventory(), false)
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		resp.InventoryFeatures = imagescan.ConvertFeatures(layer.Features)
```

Comment:
hm maybe we should put this in a more common package so we don't mix image and node scanning libraries

Reply:
We could - we kind of intentionally planned to call the imagescan package in the design doc, so having it here is a consequence and it is visible (maybe that is good?).

I would rely here on your suggestion here if you have some particular pkg in mind.

Reply:
I moved it into a separate package now. Many existing pkgs caused import cycle, so I created a separate one in e693353e46875ad3f44dc2fd388dfea1edbe080e